SpatialGDK/Source/SpatialGDK/Public/EngineClasses/Components/SpatialPingComponent.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/ActorComponent.h"
+
+#include "SpatialPingComponent.generated.h"
+
+/*
+ Offers a configurable means of measuring round-trip latency in SpatialOS deployments.
+ This component should be attached to a player controller.
+ */
+UCLASS(ClassGroup = (SpatialGDK), Meta = (BlueprintSpawnableComponent))
+class SPATIALGDK_API USpatialPingComponent : public UActorComponent
+{
+	GENERATED_UCLASS_BODY()
+
+public:
+	// Determines whether the component starts with ping behavior enabled.
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = SpatialPing)
+	bool bStartWithPingEnabled = true;
+	// The minimum time, in seconds, between pings.
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = SpatialPing)
+	float MinPingInterval = 0.5f;
```

Comment:
What's the reasoning for this being 0.5 at the moment?

Reply:
This is entirely arbitrary. It just seemed like a good number for testing it. Is there some kind of standard interval length for sending pings?

Reply:
the default for linux ping is 1sec. (https://linux.die.net/man/8/ping)